### DISCUSSION TITLE
File Upload: When uploading files, how to specify the engine (such as tpm2tss)?

### DISCUSSION THREAD

**Discussion by xunmenging** [Azure/azure-iot-sdk-c#2437] [Q&A] [answered]
I use the interface below to specify the engine, but it doesn't work for file uploads (IoTHubDeviceClient_LL_UploadMultipleBlocksToBlob).

sample code: 
static const OPTION_OPENSSL_KEY_TYPE x509_key_from_engine = KEY_TYPE_ENGINE;
static const char* opensslEngine = "tpm2tss";
Prov_Device_LL_SetOption(m_userCtx.registerHandle, OPTION_OPENSSL_ENGINE, opensslEngine);
Prov_Device_LL_SetOption(m_userCtx.registerHandle, OPTION_OPENSSL_PRIVATE_KEY_TYPE, &x509_key_from_engine);
IoTHubDeviceClient_LL_SetOption(m_userCtx.iothubHandle, OPTION_OPENSSL_ENGINE, opensslEngine);
IoTHubDeviceClient_LL_SetOption(m_userCtx.iothubHandle, OPTION_OPENSSL_PRIVATE_KEY_TYPE, &x509_key_from_engine);

**Comment by CIPop** [comment, marked as answer]
@xunmenging, this should work thanks to @JonathanKnam 's contribution: https://github.com/Azure/azure-iot-sdk-c/pull/2400

Please make sure you have this recent commit.
